File: beautarr.py
#works, but faster version in c++ (doesn't work?)
def count_beautiful_subarrays(A, N):
    count = 0
    freq = [0] * (max(A) + 1)

    l = 0
    for r in range(N):
        freq[A[r]] += 1
        while freq[A[r]] > 2:
            freq[A[l]] -= 1
            l += 1
        count += r - l + 1
    return count
N = int(input())
A = [int(i) for i in input().split()]
print(count_beautiful_subarrays(A, N))

File: beautarr.cpp
#include <iostream>
#include <map>
using namespace std;
#define int long long
int a[(int)5e5 + 5];
int freq[(int)5e5 + 5];

signed main() {
    int n;
    cin >> n;
    for (int i = 1; i <= n; i++) {
        cin >> a[i];
    }
    int l = 1, count = 0;
    for (int r = 1; r <= n; r++) {
        freq[a[r]] += 1;
        while (freq[a[r]] > 2) {
            freq[a[l]] -= 1;
            l += 1;
        }
        count += r - l + 1;
    }
    cout << count << endl;
    return 0;
}
